2025-03-27 13:31:48,238 INFO    Thread-6 (_run_job):74893 [wandb_setup.py:_flush():67] Current SDK version is 0.19.8
2025-03-27 13:31:48,238 INFO    Thread-6 (_run_job):74893 [wandb_setup.py:_flush():67] Configure stats pid to 74893
2025-03-27 13:31:48,238 INFO    Thread-6 (_run_job):74893 [wandb_setup.py:_flush():67] Loading settings from /home/gametheorylab/.config/wandb/settings
2025-03-27 13:31:48,238 INFO    Thread-6 (_run_job):74893 [wandb_setup.py:_flush():67] Loading settings from /home/gametheorylab/Desktop/RL-DA6400-assignment_1/cartpole-v1/wandb/settings
2025-03-27 13:31:48,239 INFO    Thread-6 (_run_job):74893 [wandb_setup.py:_flush():67] Loading settings from environment variables
2025-03-27 13:31:48,239 INFO    Thread-6 (_run_job):74893 [wandb_init.py:setup_run_log_directory():647] Logging user logs to /home/gametheorylab/Desktop/RL-DA6400-assignment_1/cartpole-v1/wandb/run-20250327_133148-ov2a8xej/logs/debug.log
2025-03-27 13:31:48,239 INFO    Thread-6 (_run_job):74893 [wandb_init.py:setup_run_log_directory():648] Logging internal logs to /home/gametheorylab/Desktop/RL-DA6400-assignment_1/cartpole-v1/wandb/run-20250327_133148-ov2a8xej/logs/debug-internal.log
2025-03-27 13:31:48,239 INFO    Thread-6 (_run_job):74893 [wandb_init.py:init():761] calling init triggers
2025-03-27 13:31:48,239 INFO    Thread-6 (_run_job):74893 [wandb_init.py:init():766] wandb.init called with sweep_config: {'alpha': 0.29149076714276667, 'temperature': 0.3646443944154494}
config: {'_wandb': {}}
2025-03-27 13:31:48,239 INFO    Thread-6 (_run_job):74893 [wandb_init.py:init():784] starting backend
2025-03-27 13:31:48,239 INFO    Thread-6 (_run_job):74893 [wandb_init.py:init():788] sending inform_init request
2025-03-27 13:31:48,242 INFO    Thread-6 (_run_job):74893 [backend.py:_multiprocessing_setup():101] multiprocessing start_methods=fork,spawn,forkserver, using: spawn
2025-03-27 13:31:48,242 INFO    Thread-6 (_run_job):74893 [wandb_init.py:init():798] backend started and connected
2025-03-27 13:31:48,243 INFO    Thread-6 (_run_job):74893 [wandb_run.py:_config_callback():1261] config_cb None None {'alpha': 0.29149076714276667, 'temperature': 0.3646443944154494}
2025-03-27 13:31:48,245 INFO    Thread-6 (_run_job):74893 [wandb_init.py:init():891] updated telemetry
2025-03-27 13:31:48,259 INFO    Thread-6 (_run_job):74893 [wandb_init.py:init():915] communicating run to backend with 90.0 second timeout
2025-03-27 13:31:48,717 INFO    Thread-6 (_run_job):74893 [wandb_init.py:init():990] starting run threads in backend
2025-03-27 13:31:48,858 INFO    Thread-6 (_run_job):74893 [wandb_run.py:_console_start():2375] atexit reg
2025-03-27 13:31:48,858 INFO    Thread-6 (_run_job):74893 [wandb_run.py:_redirect():2227] redirect: wrap_raw
2025-03-27 13:31:48,858 INFO    Thread-6 (_run_job):74893 [wandb_run.py:_redirect():2292] Wrapping output streams.
2025-03-27 13:31:48,858 INFO    Thread-6 (_run_job):74893 [wandb_run.py:_redirect():2315] Redirects installed.
2025-03-27 13:31:48,858 INFO    Thread-6 (_run_job):74893 [wandb_init.py:init():1032] run started, returning control to user process
2025-03-27 13:31:48,859 INFO    Thread-6 (_run_job):74893 [wandb_run.py:_finish():2112] finishing run ee24e009-iitm/qlearning-cartpole/ov2a8xej
2025-03-27 13:31:48,859 INFO    Thread-6 (_run_job):74893 [wandb_run.py:_atexit_cleanup():2340] got exitcode: 1
2025-03-27 13:31:48,859 INFO    Thread-6 (_run_job):74893 [wandb_run.py:_restore():2322] restore
2025-03-27 13:31:48,859 INFO    Thread-6 (_run_job):74893 [wandb_run.py:_restore():2328] restore done
2025-03-27 13:31:49,181 INFO    MsgRouterThr:74893 [mailbox.py:close():129] Closing mailbox, abandoning 2 handles.
2025-03-27 13:31:49,182 INFO    Thread-6 (_run_job):74893 [wandb_run.py:_restore():2322] restore
2025-03-27 13:31:49,182 INFO    Thread-6 (_run_job):74893 [wandb_run.py:_restore():2328] restore done
2025-03-27 13:31:49,182 ERROR   Thread-6 (_run_job):74893 [wandb_run.py:_atexit_cleanup():2361] Problem finishing run
Traceback (most recent call last):
  File "/home/gametheorylab/miniconda3/lib/python3.12/site-packages/wandb/agents/pyagent.py", line 306, in _run_job
    self._function()
  File "/home/gametheorylab/Desktop/RL-DA6400-assignment_1/cartpole-v1/wandb_qlearning.py", line 23, in train
    rewards = run_qlearning(
              ^^^^^^^^^^^^^^
  File "/home/gametheorylab/Desktop/RL-DA6400-assignment_1/cartpole-v1/q_runner.py", line 14, in run_qlearning
    q_table, bins = Qtable(env.observation_space, env.action_space, num_bins)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gametheorylab/Desktop/RL-DA6400-assignment_1/cartpole-v1/q_table.py", line 24, in Qtable
    bins[2] = np.linspace(-0.42, 0.42, bin_size)
    ~~~~^^^
IndexError: index 2 is out of bounds for axis 0 with size 2

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/gametheorylab/miniconda3/lib/python3.12/site-packages/wandb/sdk/wandb_run.py", line 2352, in _atexit_cleanup
    self._on_finish()
  File "/home/gametheorylab/miniconda3/lib/python3.12/site-packages/wandb/sdk/wandb_run.py", line 2609, in _on_finish
    wait_with_progress(
  File "/home/gametheorylab/miniconda3/lib/python3.12/site-packages/wandb/sdk/mailbox/wait_with_progress.py", line 24, in wait_with_progress
    return wait_all_with_progress(
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gametheorylab/miniconda3/lib/python3.12/site-packages/wandb/sdk/mailbox/wait_with_progress.py", line 70, in wait_all_with_progress
    return _wait_handles(handle_list, timeout=progress_after)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gametheorylab/miniconda3/lib/python3.12/site-packages/wandb/sdk/mailbox/wait_with_progress.py", line 110, in _wait_handles
    results.append(handle.wait_or(timeout=remaining_timeout))
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gametheorylab/miniconda3/lib/python3.12/site-packages/wandb/sdk/mailbox/mailbox_handle.py", line 122, in wait_or
    return self._fn(self._handle.wait_or(timeout=timeout))
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gametheorylab/miniconda3/lib/python3.12/site-packages/wandb/sdk/mailbox/response_handle.py", line 98, in wait_or
    raise HandleAbandonedError()
wandb.sdk.mailbox.mailbox_handle.HandleAbandonedError
